src/Features/Core/Portable/Completion/Providers/ImportCompletionProvider/AbstractTypeImportCompletionService.cs

```diff
@@ -372,15 +381,52 @@ public bool IsGeneric
             public bool IsAttribute
                 => (_properties & ItemPropertyKind.IsAttribute) != 0;
 
+            public bool IsEditorBrowsableStateAdvanced
+                => (_properties & ItemPropertyKind.IsEditorBrowsableStateAdvanced) != 0;
+
             public TypeImportCompletionItemInfo WithItem(CompletionItem item)
-                => new TypeImportCompletionItemInfo(item, IsPublic, IsGeneric, IsAttribute);
+                => new TypeImportCompletionItemInfo(item, IsPublic, IsGeneric, IsAttribute, IsEditorBrowsableStateAdvanced);
 
             [Flags]
             private enum ItemPropertyKind : byte
             {
                 IsPublic = 0x1,
                 IsGeneric = 0x2,
                 IsAttribute = 0x4,
+                IsEditorBrowsableStateAdvanced = 0x8,
+            }
+        }
+
+        // Things needed for determining whether a symbol is EditorBrowsable.
+        // Grouped together and reused within each compilation.
+        private class EditorBrowsableInfo
+        {
+            private Optional<INamedTypeSymbol?>? _hideModuleNameAttribute;
+            private Optional<IMethodSymbol?>? _editorBrowsableAttributeConstructor;
+            private ImmutableArray<IMethodSymbol>? _typeLibTypeAttributeConstructors;
+            private ImmutableArray<IMethodSymbol>? _typeLibFuncAttributeConstructors;
+            private ImmutableArray<IMethodSymbol>? _typeLibVarAttributeConstructors;
```

Comment:
ok.  being lazy doesn't seem helpful at all given: https://github.com/dotnet/roslyn/pull/47771/files#diff-e288fb559f423bd4c1aef8cae5d9daf2R150

don't be lazy here.  just initialize this stuff in the constructor and pass this all around as a fully realized entity.

Reply:
That's not true. We won't need to initialize those if we end up having a cache hit.

Reply:
As discussed offline, the cost of this after removing laziness is high. So I keep it this way with some additional comments